config/routes.rb

```diff
@@ -1,6 +1,17 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+	resources :dishes, only: [:show, :update] do
+		# Nested route for ingredients for a dish
+		resources :ingredients, only: [:index, :create]
+		resources :dish_ingredients, only: [:create]
```

Comment:
For the route for adding an ingredient to a dish, you could instead just do
```
resources :dishes, only: :show do
      resources :ingredients, only: :create, controller: "dish_ingredients"
end
```
This covers all the routes needed for US 1 and 2. Notice how the ingredients controller nested under there can direct to a joins controller (see comment in controllers about the RESTful design of this action)